### PR TITLE
@alloy => [Autosuggest] Allow QueryRenderer to return w/o fetching

### PR DESCRIPTION
**Pull request by mzikherman**
This allows the SearchBar to start off initially w/o making an unnecessary round trip to MP (requesting `viewer {}`), since that's unneeded for the initial state.

This was based off of a discussion in the [Relay Slack chat](https://graphql.slack.com/archives/C0BEXJLKG/p1550598752174300), with this solution (modifying the network layer) seeming to be the most workable.

### CODE REVIEW

**Comment by artsyit** [comment]
Deploy preview for *artsy-reaction* ready!

Built with commit 61c570a0a10cce50a43aba314000cd262e093591

https://deploy-preview-2039--artsy-reaction.netlify.com

File: src/Artsy/Relay/createRelaySSREnvironment.ts [hunk 1]
```diff
@@ -74,6 +74,14 @@ export function createRelaySSREnvironment(config: Config = {}) {
   const network =
     relayNetwork ||
     new RelayNetworkLayer([
+      // TODO: Better introspection around if this is a SearchBar query,
+      // or further refactoring to extract `addMiddlewareToEnvironment(environment)`,
+      // to be used in the SearchBar QueryRenderer (for example).
+      next => req => {
+        if (req.variables.term === "")
```
Comment (alloy) [inline src/Artsy/Relay/createRelaySSREnvironment.ts]: I have a few other things on my plate that I want to get done first, so can you file a ticket for allowing to scope this to just a single QueryRenderer and assign it to me? In the meantime you can perhaps ensure this short-circuiting only applies to the right query by adding a check for something like `req.query.includes("SearchBarQuery")`

Comment (mzikherman) [reply src/Artsy/Relay/createRelaySSREnvironment.ts]: Will do!

Comment (mzikherman) [reply src/Artsy/Relay/createRelaySSREnvironment.ts]: Turns out `req` isn't the usual 'express/node' request object, but a `RelayRequest`, so `req.id` is the name of the query `SearchSuggestQuery`, doesn't even need an `includes`!

**Comment by artsyit** [comment]
:rocket: PR was released in v14.0.6 :rocket:

File: src/Artsy/Relay/createRelaySSREnvironment.ts (hunk 1 above)
Comment (alloy) [reply src/Artsy/Relay/createRelaySSREnvironment.ts]: Perfect 👌

**Pull request merged**